### PR TITLE
cachingmap pkg

### PR DESCRIPTION
**Pull request by tomastigera**
## Description
    felix: cachingmap is BPF-independent package

    felix: caching map is BPF independent

    felix: cachingmap test in _test package

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
TBD
```

## Reminder for the reviewer

Make sure that this PR has the correct labels and milestone set.

Every PR needs one `docs-*` label.

- `docs-pr-required`: This change requires a change to the documentation that has not been completed yet.
- `docs-completed`: This change has all necessary documentation completed.
- `docs-not-required`: This change has no user-facing impact and requires no docs.

Every PR needs one `release-note-*` label.

- `release-note-required`: This PR has user-facing changes. Most PRs should have this label.
- `release-note-not-required`: This PR has no user-facing changes.

Other optional labels:

- `cherry-pick-candidate`: This PR should be cherry-picked to an earlier release. For bug fixes only.
- `needs-operator-pr`: This PR is related to install and requires a corresponding change to the operator.

### CODE REVIEW

File: felix/cachingmap/caching_map.go [hunk 1]
```diff
@@ -260,16 +264,16 @@ func (c *CachingMap[K, V]) ApplyUpdatesOnly() error {
...
 	err := c.maybeLoadCache()
 	if err != nil {
 		return err
 	}
 	var errs ErrSlice
 	for k := range c.pendingDeletions {
 		err := c.dpMap.Delete(k)
-		if err != nil && !bpf.IsNotExists(err) {
-			logrus.WithError(err).Warn("Error while deleting from BPF map")
+		if err != nil && err != ErrNotExists {
```
Comment (fasaxc) [inline felix/cachingmap/caching_map.go:275]: Annoying little problem dealing with different exception types that mean the same thing.  Wonder if we should pass in an `shouldIgnoreDeletionError(error) bool` func or maybe change the code we use it with to not return a not exists error?

**Review by fasaxc — CHANGES_REQUESTED** [review]
Just one thought that the change to error handling might bite us if we use it with some other underlying map later (or if we refactor the BPF map to return say, a wrapped error).

File: felix/cachingmap/caching_map.go (hunk 1 above)
Comment (tomastigera) [reply felix/cachingmap/caching_map.go:275]: This gave me a hard time, and I originally though (and forgot to do) that the bpf.TypedMap would return this error on behalf of the bpf.Map. And that any other user would return this error as well. But I like the idea that the cachingmap.DataplaneMap would have `ShouldIgnoreDeletionError` or `IsEExist()` or something like that.

**Review by fasaxc — APPROVED** [review]
Ship it.

**Pull request merged**